instrumentation/kamon-armeria/src/main/java/kamon/instrumentation/armeria/server/ArmeriaHttpServerDecorator.java

```diff
@@ -0,0 +1,72 @@
+/* =========================================================================================
+ * Copyright © 2013-2020 the kamon project <http://kamon.io/>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License") you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ * =========================================================================================
+ */
+
+package kamon.instrumentation.armeria.server;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.SimpleDecoratingHttpService;
+import io.netty.util.AttributeKey;
+import kamon.Kamon;
+import kamon.context.Storage;
+import kamon.instrumentation.armeria.converters.KamonArmeriaMessageConverter;
+import kamon.instrumentation.http.HttpServerInstrumentation;
+
+import java.util.Map;
+
+public class ArmeriaHttpServerDecorator extends SimpleDecoratingHttpService {
+  public static final AttributeKey<HttpServerInstrumentation.RequestHandler> REQUEST_HANDLER_TRACE_KEY =
+      AttributeKey.valueOf(HttpServerInstrumentation.RequestHandler.class, "REQUEST_HANDLER_TRACE");
+
+  private final Map<Integer, HttpServerInstrumentation> serverInstrumentationMap;
+
+  public ArmeriaHttpServerDecorator(HttpService delegate, Map<Integer, HttpServerInstrumentation> serverInstrumentations) {
+    super(delegate);
+    this.serverInstrumentationMap = serverInstrumentations;
+  }
+
+  @Override
+  public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+    final HttpServerInstrumentation httpServerInstrumentation = serverInstrumentationMap.get(req.uri().getPort());
+
+    if (httpServerInstrumentation != null) {
+
+      final HttpServerInstrumentation.RequestHandler requestHandler =
```

Comment:
I was trying to implement this suggested change like this

```java
      ctx.log()
          .whenComplete()
          .thenAccept(log -> {
              final Context context = ctx.attr(REQUEST_HANDLER_TRACE_KEY).context();
              requestHandler.span().name(ctx.config().route().patternString());
              requestHandler.buildResponse(KamonArmeriaMessageConverter.toResponse(log),context);
              requestHandler.responseSent();
          });
```

And I've some concerns about this:
- If a route does not exist in the server we will potentially generate N operation names (1 per route that does not exist in the server), because `ctx.config().route().patternString()` will return the request path even if the route does not exist. The current implementation ensures that the operation name will be set to `unhandled`(because of [this](https://github.com/kamon-io/Kamon/pull/854/files#diff-2343eaa073a7618196380a20748e653f9ad2f420f42c35d3825f2f5838bcc01dR131)) in this case. Probably we can check if we already set this operation name to the configured value for unhandled requests. I was wondering which will be a better approach, what do you think?
- The client context, `ClientRequestContext`, does not have a method like `ctx.config().route().patternString()` so we'll need the operation name generator, right? I understand that users could have cardinality issues with the current implementation if they don't set operation mappings but I think there is nothing we can do here

Reply:
Where we at here?

Reply:
We are using the  `patternString()` method provided by Armeria to generate operation names only in the server side, and we set the operation name as `unhandled` by default (it's configurable) if the route wasn't found.

In the client side we  still use the custom operation name generator because we don't have an equivalent `patternString()` method. at least that is true until the version of Armeria that we are currently using which is not the latest.